### PR TITLE
[RW-12046][risk=no] PD maybe resized down without warning causing user's data to be deleted

### PR DESCRIPTION
**Pull request by NehaBroad**
As part of this PR:

- Call `updateRuntime` to increase disk size instead of `updateDisk`, as `diskApi.updateDisk` only updated the size in the database without informing the runtime of the change. By calling `runtimeApi.updateRuntime`, not only will the disk size be updated, but it will also ensure that the runtime is aware of it.

- Update the `updateRuntime` object to remove the `PD` object."

Follow up PR:

- There are 2 test left in enzyme that needs to be corrected

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx [hunk 1]
```diff
@@ -1161,10 +491,13 @@ describe(RuntimeConfigurationPanel.name, () => {
       await clickButtonIfVisible(wrapper, 'Delete');
     }
 
+    console.log('update disk' + setters);
```
Comment (evrii) [inline ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:494]: I think that these logs might have gotten left behind on accident.

File: ui/src/app/utils/runtime-hooks.tsx [hunk 2]
```diff
@@ -147,6 +147,11 @@ export const useDisk = (currentWorkspaceNamespace: string) => {
   }, [currentWorkspaceNamespace]);
 };
 
+let aborter = new AbortController();
+
+// This function is being used by test ONLY to abort the runtime polling so that the tests do not interfere with each
+// other
+export const getAborter = () => aborter;
 // useRuntimeStatus hook can be used to change the status of the runtime
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-hooks.tsx:155]: add a new line here, so it's clear that this is not related to the aborter

File: ui/src/app/utils/runtime-hooks.tsx [hunk 3]
```diff
@@ -190,13 +195,15 @@ export const useRuntimeStatus = (
         () => (r) => r.status === RuntimeStatus.STOPPED,
       ]
     );
+    aborter = new AbortController();
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-hooks.tsx:198]: why is this necessary?  it is set in L150

File: ui/src/app/utils/runtime-hooks.tsx [hunk 4]
```diff
@@ -292,7 +299,7 @@ export const useCustomRuntime = (
 
   useEffect(() => {
     let mounted = true;
-    const aborter = new AbortController();
+    aborter = new AbortController();
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-hooks.tsx:302]: same comment: why is this necessary?  it is set in L150

File: ui/src/app/utils/runtime-hooks.tsx [hunk 5]
```diff
@@ -311,15 +318,6 @@ export const useCustomRuntime = (
           )
         );
 
-        // A disk update may be need in combination with a runtime update.
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-hooks.tsx:314]: why did this logic move to a different location?

File: ui/src/testing/react-test-helpers.tsx [hunk 6]
```diff
@@ -204,6 +204,13 @@ export const expectDropdownDisabled = (
   ).not.toBeInTheDocument();
 };
 
+export const getDropdownSelection = (
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/testing/react-test-helpers.tsx:214]: My guess is that this is difficult because it's PrimeReact?

But I have a different comment: we have `id` for these elements that can be used here.  Is it necessary to duplicate this with `name` ?

File: ui/src/app/components/runtime-configuration-panel/offer-delete-disk-with-update.tsx [hunk 7]
```diff
@@ -86,6 +86,7 @@ export const OfferDeleteDiskWithUpdate = ({
               style={{ display: 'inline-block', marginRight: '0.75rem' }}
             >
               <RadioButton
+                aria-label={'delete-pd'}
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel/offer-delete-disk-with-update.tsx:89]: This is not an appropriate ARIA label.  These should be formatted to be readable by users, not machines.  So a better choice here would be "Delete Persistent Disk"

File: ui/src/app/utils/runtime-hooks.tsx (hunk 5 above)
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx:314]: I was placing the updateRuntimes conditions together for making it easy to read, i can move it back

File: ui/src/testing/react-test-helpers.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/testing/react-test-helpers.tsx:214]: The prime react dropdown has select in the following way beause of which i have to use name
eg
<select
                  name="runtime-cpu"
                  tabindex="-1"
                  aria-hidden="true"
                  data-pc-section="select"
                >

File: ui/src/app/utils/runtime-hooks.tsx (hunk 4 above)
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx:302]: This causes interference in the test if its not declared again

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 8]
```diff
@@ -199,15 +199,19 @@ private LeonardoCreateRuntimeRequest buildCreateRuntimeRequest(
...
+  private Object buildRuntimeConfig(Runtime runtime, boolean updateRuntime) {
     if (runtime.getGceConfig() != null) {
       return leonardoMapper
           .toLeonardoGceConfig(runtime.getGceConfig())
           .zone(workbenchConfigProvider.get().firecloud.gceVmZone);
     } else if (runtime.getGceWithPdConfig() != null) {
-      return leonardoMapper
-          .toLeonardoGceWithPdConfig(runtime.getGceWithPdConfig())
-          .zone(workbenchConfigProvider.get().firecloud.gceVmZone);
+      return updateRuntime
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java]: This has been updated, we are now using different objects for updateruntime

File: ui/src/app/components/runtime-configuration-panel/offer-delete-disk-with-update.tsx (hunk 7 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel/offer-delete-disk-with-update.tsx]: Done

File: ui/src/app/utils/runtime-hooks.tsx (hunk 2 above)
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx:156]: Done

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 9]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    await pickMainCpu(container, 8);
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(MIN_DISK_SIZE_GB + 10);
+
+    // GPU
+    await pickPresets(container, runtimePresets.generalAnalysis.displayName);
+
+    clickExpectedButton('Create');
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual('Creating');
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:1455]: don't need to change it for this PR, but best practice is to include only 1 expect() in a waitFor() - then use expect() without waitFor() for the others.  the logic is that you have already waited at that point.

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx [hunk 10]
```diff
@@ -303,46 +275,6 @@ describe(RuntimeConfigurationPanel.name, () => {
...
-
-  it('should allow creation with GCE with PD config', async () => {
-    // migrated to RTL
-  });
-
-  it('should allow creation with Dataproc config', async () => {
-    // migrated to RTL
-  });
-
   it('should allow configuration via GCE preset', async () => {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:278]: also line 376

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 11]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+        expect(runtimeApiStub.runtime.status).toEqual('Creating');
+        expect(runtimeApiStub.runtime.configurationType).toEqual(
+          RuntimeConfigurationType.HAIL_GENOMIC_ANALYSIS
+        );
+        expect(runtimeApiStub.runtime.dataprocConfig).toEqual(
+          runtimePresets.hailAnalysis.runtimeTemplate.dataprocConfig
+        );
+        expect(runtimeApiStub.runtime.gceConfig).toBeFalsy();
+      },
+      { interval: 750 }
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:1494]: don't need this - I tried it

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 12]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+      expect(customDiskSize).not.toEqual(persistentDisk.size);
+      const { container } = component();
+      // Try :Check there is no spinner
+      expect(getMainCpu(container)).toEqual(
+        findMachineByName(machineType).cpu.toString()
+      );
+      expect(getMainRam(container)).toEqual(
+        findMachineByName(machineType).memory.toString()
+      );
+      expect(getDetachableDiskValue()).toEqual('120');
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:1531]: it's not obvious what 120 represents here, so it should be `persistentDisk.size`

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 13]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    await pickNumWorkers(10);
+    await pickNumPreemptibleWorkers(20);
+    await pickPresets(container, runtimePresets.hailAnalysis.displayName);
+    clickExpectedButton('Create');
+
+    await waitFor(
+      () => {
+        expect(runtimeApiStub.runtime.status).toEqual('Creating');
+      },
+      { timeout: 4000 }
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:1632]: don't need this

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx [hunk 14]
```diff
@@ -398,106 +330,20 @@ describe(RuntimeConfigurationPanel.name, () => {
...
-      },
-      dataprocConfig: null,
-    });
-
-    const wrapper = await component();
-    expect(getDetachableDiskSize(wrapper)).toEqual(disk.size);
+    // Moved to RTL
   });
 
   it('should allow configuration via dataproc preset from modified form', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:349]: mark this and the following 2 tests as converted

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 15]
```diff
@@ -1223,4 +1340,864 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    setCurrentRuntime({
+      ...runtimeApiStub.runtime,
+      gceConfig: null,
+      gceWithPdConfig: null,
+      dataprocConfig: defaultDataprocConfig(),
+      status: RuntimeStatus.CREATING,
+    });
+
+    await component();
+
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2018]: yes I think so.  As I understand it, the original test wants to show what happend when:
* runtime is Creating
* user made a change

So we're missing the second part

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 16]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    expect(getRunningCost()).toEqual('$0.87 per hour');
+    expect(getPausedCost()).toEqual('$0.13 per hour');
+  });
+
+  it('should prevent runtime update when disk size is invalid', async () => {
+    const { container } = await component();
+
+    const getNextButton = () => screen.getByRole('button', { name: 'Next' });
+
+    await pickDetachableDiskSize(49);
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2182]: We can remove the detachable-disks part of this test, because it's covered in the next test.

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx [hunk 17]
```diff
@@ -1243,176 +691,43 @@ describe(RuntimeConfigurationPanel.name, () => {
...
-      const wrapper = await component();
-      await mustClickButton(wrapper, 'Customize');
-      await runDetachableDiskCase(
-        wrapper,
-        [desc, setters, expectations],
-        disk.name
-      );
-    }
-  );
+  // test.each([
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:694]: can remove this as well as all the "detachableDiskCase" stuff (from this file)

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 18]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    clickExpectedButton('Update');
+    await waitFor(async () => {
+      runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+      expect(runtimeApiStub.runtime.gceWithPdConfig).not.toBeNull();
+    });
+    await waitFor(() => {}, { interval: 1000 });
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(runtimeApiStub.runtime.gceWithPdConfig).toBeUndefined();
+      // expect(runtimeApiStub.runtime.gceWithPdConfig).toBeNull();
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2339]: rm

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 19]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\. /i
+      )
+    ).toBeTruthy();
+
+    togglePDRadioButton();
+
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    await waitFor(async () => {
+      runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2332]: this can go outside the waitFor

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 20]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+      expect(runtimeApiStub.runtime.gceWithPdConfig).not.toBeNull();
+    });
+    await waitFor(() => {}, { interval: 1000 });
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2335]: don't need this

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 21]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    clickExpectedButton('Update');
+    runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(disksApiStub.disk?.name).toEqual(disk.name);
+    });
+  });
+
+  it('should prevent runtime creation when running cost is too high for free tier', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2370]: pls rename "free tier" to "initial credits"

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 22]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    createButton = screen.getByRole('button', { name: 'Create' });
+    // We don't want to disable for user provided billing. Just put a warning.
+    expectButtonElementEnabled(createButton);
+    expect(
+      screen.findByText(
+        'Your runtime is expensive. Are you sure you wish to proceed?'
+      )
+    ).toBeTruthy();
+
+    await pickNumWorkers(2);
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2453]: not sure why this part of the test is needed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 23]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(1);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should allow runtime creation when running cost is too high for user provided billing', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2404]: this test seems redundant.  it's included in the next test.

**Review by jmthibault79 — COMMENTED** [review]
Lots of great test conversions here!  Thank you for doing this work that I was too lazy to do ;-)

I have a few requests for changes, but we're really close now.

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 16 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2174]: I want to confirm if i understand the change correct, we can discuss this in mob

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 23 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Yeah wasnt sure about it thought presence of billingAccountName: 'user provided billing' made a different. Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 24]
```diff
@@ -631,12 +643,64 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+          labels: {},
+        },
+        gpuConfig: null,
+      },
+      gceConfig: null,
+      dataprocConfig: null,
+    };
+  };
+
+  type DetachableDiskCase = [
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Moved

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 25]
```diff
@@ -997,19 +1210,17 @@ describe(RuntimeConfigurationPanel.name, () => {
...
-    await pickMainCpu(container, user, 8);
-    await pickMainRam(container, user, 52);
-    await pickDetachableDiskSize(user, MIN_DISK_SIZE_GB + 10);
+    await pickMainCpu(container, 8);
+    await pickMainRam(container, 52);
+    await pickDetachableDiskSize(MIN_DISK_SIZE_GB + 10);
 
     clickExpectedButton('Create');
-
+    jest.advanceTimersByTime(1000);
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 9 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:1454]: i tried removing it here and all the other places

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 11 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 26]
```diff
@@ -1223,4 +1340,864 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+
+      // show that the preset values do not match the existing runtime
+
+      const { machineType, persistentDisk } =
+        runtimePresets.generalAnalysis.runtimeTemplate.gceWithPdConfig;
+
+      expect(customMachineType).not.toEqual(machineType);
+      expect(customDiskSize).not.toEqual(persistentDisk.size);
+      const { container } = component();
+      // Try :Check there is no spinner
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 12 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Updated

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 13 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 19 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed waitFor for Deleted status

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 20 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 18 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Removed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 21 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: renamed

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 27]
```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    const workerCountInput: HTMLInputElement = spinDiskElement('num-workers');
+    expect(workerCountInput.disabled).toBeTruthy();
+
+    const preemptibleCountInput: HTMLInputElement =
+      spinDiskElement('num-preemptible');
+    expect(preemptibleCountInput.disabled).toBeTruthy();
+  });
+
+  jest.setTimeout(50000);
+  test.skip.each([
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2448]: Added comment, we can discuss if this is fine or not

File: ui/src/app/utils/runtime-hooks.tsx (hunk 4 above)
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx:303]: Yeah the only thing i know is that every time we render useCustome is triggered which calls the useEffect where its reinitializes. We can debug this in mob?

File: ui/src/app/utils/runtime-hooks.tsx (hunk 5 above)
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx:314]: Done

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:234]: Done

File: ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx (hunk 14 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.enzyme.spec.tsx:262]: Done

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2010]: So re-reading the test: if the runtime is in creating state all the options are disabled along with Next button.  I am not able to understand how the made a change

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 22 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2427]: I updated it to say if the number of worker is within range remove the banner. If that condition doesnt make sense i can remove it

Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2427]: Basically saying the banner does not stick and can be reset

File: ui/src/app/utils/runtime-hooks.tsx [hunk 28]
```diff
@@ -311,15 +319,21 @@ export const useCustomRuntime = (
...
         if (mostSevereDiskDiff === AnalysisDiffState.CAN_UPDATE_IN_PLACE) {
-          await disksApi().updateDisk(
-            currentWorkspaceNamespace,
-            existingDisk.name,
-            requestedDisk.size
-          );
+          await runtimeApi().updateRuntime(currentWorkspaceNamespace, {
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/utils/runtime-hooks.tsx]: Done

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2010]: Updated

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 16 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx]: Moved to another test

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx (hunk 22 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2427]: ok great, this works

File: ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx [hunk 29]
```diff
@@ -1223,4 +1429,1134 @@ describe(RuntimeConfigurationPanel.name, () => {
...
+    // This should make the cost around $160 per hour.
+    await pickNumWorkers(700);
+    createButton = screen.getByRole('button', { name: 'Create' });
+    // We don't want to disable for user provided billing. Just put a warning.
+    expectButtonElementEnabled(createButton);
+    expect(
+      screen.findByText(
+        'Your runtime is expensive. Are you sure you wish to proceed?'
+      )
+    ).toBeTruthy();
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx:2424]: nit: I'd prefer `.toBeInTheDocument()` / `.not.toBeInTheDocument()` for cases like this

**Review by jmthibault79 — APPROVED** [review]
Tested locally.  We can no longer get into the state that causes the bug!
